src/com/seafile/seadroid2/ui/adapter/SeafItemAdapter.java

```diff
@@ -284,41 +290,17 @@ private void setFileView(SeafDirent dirent, Viewholder viewHolder, int position)
 
             viewHolder.downloadStatusIcon.setImageResource(downloadStatusIcon);
             viewHolder.subtitle.setText(dirent.getSubtitle());
+        }
+
+        if (Utils.isViewableImage(file.getName())) {
```

Comment:
or use a method, like `Utils.isThumbnailExist`, to avoid repeatedly downloading.